brody_lab_to_nwb/interfaces/protocol_info/protocol_info_utils.py

```diff
@@ -0,0 +1,192 @@
+import numpy as np
+import pandas as pd
+import scipy.io as spio
+
+
+def load_nested_mat(filename):
+    """
+    Replace scipy.io.loadmat.
+
+    It cures the problem of not properly recovering python dictionaries from mat files.
+    It calls the function check keys to cure all entries which are still mat-objects.
+    From https://stackoverflow.com/questions/48970785/complex-matlab-struct-mat-file-read-by-python.
+    """
+
+    def _check_vars(d):
+        """
+        Check if entries in dictionary are mat-objects.
+
+        If yes todict is called to change them to nested dictionaries
+        """
+        for key in d:
+            if isinstance(d[key], spio.matlab.mio5_params.mat_struct):
+                d[key] = _todict(d[key])
+            elif isinstance(d[key], np.ndarray):
+                d[key] = _toarray(d[key])
+        return d
+
+    def _todict(matobj):
+        """Recursive function which constructs from matobjects nested dictionaries."""
+        d = {}
+        for strg in matobj._fieldnames:
+            elem = matobj.__dict__[strg]
+            if isinstance(elem, spio.matlab.mio5_params.mat_struct):
+                d[strg] = _todict(elem)
+            elif isinstance(elem, np.ndarray):
+                d[strg] = _toarray(elem)
+            else:
+                d[strg] = elem
+        return d
+
+    def _toarray(ndarray):
+        """
+        Recursive function which constructs ndarray from cellarrays (which are loaded as numpy ndarrays).
+
+        Recurses into the elements if they contain matobjects.
+        """
+        if ndarray.dtype != 'float64':
+            elem_list = []
+            for sub_elem in ndarray:
+                if isinstance(sub_elem, spio.matlab.mio5_params.mat_struct):
+                    elem_list.append(_todict(sub_elem))
+                elif isinstance(sub_elem, np.ndarray):
+                    elem_list.append(_toarray(sub_elem))
+                else:
+                    elem_list.append(sub_elem)
+            return np.array(elem_list)
+        else:
+            return ndarray
+
+    data = spio.loadmat(filename, struct_as_record=False, squeeze_me=True)
+    return _check_vars(data)
+
+
+def make_beh_df(beh_info):
+    """
+    Wangle the ndarry from load_behaviorand and put it into a tidy dataframe.
+
+    Parameters
+    ----------
+    beh_info : ndarray
+        Extracted .mat structure.
+
+    Returns
+    -------
+    beh_df : df (ntrials x items)
+        Tidy data frame with behavior information & some relabeling.
+    """
+    beh_df = pd.DataFrame()
+    pd.options.mode.chained_assignment = None
+
+    # assign trail n values
+    beh_df['trial_num'] = np.arange(1, beh_info['n_completed_trials'] + 1)
+
+    # rename previous side to be L/R
+    prev_side_adj = np.roll(beh_info['prev_side'], 1)  # n-1 trial info
+    prev_side_adj = np.where(prev_side_adj == 114, 'RIGHT', 'LEFT')
+    prev_side_adj[0] = 'NaN'  # trial 0 doesn't have a previous
```

Comment:
Are you using the string `'NaN'` here?

Reply:
Not sure, everything in these utils is from https://github.com/Brody-Lab/jbreda_PWM_mPFC_analysis/blob/main/io_utils.py so maybe @jess-breda can explain; on the conversion side of things, I can tell you it goes into the NWBFile looking like 
![image](https://user-images.githubusercontent.com/51133164/134969942-1f97acb1-a220-4e8f-8fd0-807614edf164.png)

Reply:
OK. "NaN" means "not a number," which isn't ideal here, since the other values aren't numbers either. How about a value of "none" or "n.a." for "not applicable"?

Reply:
Set to "N/A" now